src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageAction.java

```diff
@@ -160,6 +157,43 @@ protected ActionResult execute() throws EntityDoesNotExistException {
         }
     }
 
+    private void setBundleForQuestionView(String needAjax, String courseId, String feedbackSessionName,
+                                    InstructorAttributes instructor, InstructorFeedbackResultsPageData data,
+                                    String selectedSection, String sortType, String questionNumStr)
+                                    throws EntityDoesNotExistException {
+        if (ALL_SECTION_OPTION.equals(selectedSection) && questionNumStr == null) {
+            // load page structure without responses
+            
+            data.setLargeNumberOfRespondants(needAjax != null);
+            
+            // all sections and all questions for question view
+            // set up question tables, responses to load by ajax
+            data.setBundle(logic
+                        .getFeedbackSessionResultsForInstructorWithinRangeFromView(feedbackSessionName, courseId,
+                                                                                   instructor.email,
+                                                                                   1, sortType));
+            // set isComplete to true to prevent behavior when there are too many responses, 
+            // such as the display of warning messages
+            data.getBundle().isComplete = true;
+        } else if (questionNumStr == null) {
+            // bundle for all questions, with a selected section
+            data.setBundle(logic.getFeedbackSessionResultsForInstructorInSection(feedbackSessionName, courseId,
+                                                                                instructor.email,
+                                                                                selectedSection));
+        } else if (ALL_SECTION_OPTION.equals(selectedSection) && questionNumStr != null) {
+            // bundle for a specific question, with all sections
+            int questionNum = Integer.parseInt(questionNumStr);
+            data.setBundle(logic.getFeedbackSessionResultsForInstructorFromQuestion(feedbackSessionName, courseId, 
+                                                                                   instructor.email, questionNum));
+        } else {
+            // bundle for a specific question and a specific section
+            int questionNum = Integer.parseInt(questionNumStr);
+            data.setBundle(logic.getFeedbackSessionResultsForInstructorFromQuestionInSection(
+                                            feedbackSessionName, courseId, 
+                                            instructor.email, questionNum, selectedSection));
+        }
+    }
```

Comment:
Would this be better represented as either of the following?
The current condition flow is somewhat unclear.
```
a = isAllSections
b = questionNumStr == null
```
```
if (a && b)
if (!a && b)
if (a && !b)
if (!a && !b)
```
or
```
if (b)
    if (a)
    else
else
    if (a)
    else
```